docs/rest_api.md

```diff
@@ -2,3 +2,266 @@
+You can also retrieve these credentials using the CLI running `wsk property get --auth`.
+The following is an example that uses the cURL command to get the list of all packages in the `whisk.system` namespace:
+
+```bash
+curl -u USERNAME:PASSWORD https://openwhisk.ng.bluemix.net/api/v1/namespaces/whisk.system/packages
+```
+
+```json
+[
+  {
+    "name": "slack",
+    "binding": false,
+    "publish": true,
+    "annotations": [
+      {
+        "key": "description",
+        "value": "Package that contains actions to interact with the Slack messaging service"
+      }
+    ],
+    "version": "0.0.1",
+    "namespace": "whisk.system"
+  }
+]
+```
+
+The OpenWhisk API supports request-response calls from web clients. OpenWhisk responds to `OPTIONS` requests with Cross-Origin Resource Sharing headers. Currently, all origins are allowed (that is, Access-Control-Allow-Origin is "`*`") and Access-Control-Allow-Headers yield Authorization and Content-Type.
+
+**Attention:** Because OpenWhisk currently supports only one key per account, it is not recommended to use CORS beyond simple experiments. Use [Web Actions](webactions.md) or [API Gateway](apigateway.md) to expose your actions to the public and not use the OpenWhisk authorization key for client applications that require CORS.
+
+## Using the CLI verbose mode
+
+The OpenWhisk CLI is an interface to the OpenWhisk REST API.
+You can run the CLI in verbose mode with the flag `-v`, this will  to print all the information about the HTTP request and response.
+
+Let's try getting the namespace value for the current user.
+```
+wsk namespace list -v
+```
+```
+REQUEST:
+[GET]	https://openwhisk.ng.bluemix.net/api/v1/namespaces
+Req Headers
+{
+  "Authorization": [
+    "Basic XXXYYYY"
+  ]
+}
+RESPONSE:Got response with code 200
+Resp Headers
+{
+  "Content-Type": [
+    "application/json; charset=UTF-8"
+  ]
+}
+Response body size is 28 bytes
+Response body received:
+["john@example.com_dev"]
+```
+
+As you can see you the printed information provides the properties of the HTTP request, it performs a HTTP method `GET` on the URL `https://openwhisk.ng.bluemix.net/api/v1/namespaces` using a Basic Authorization header `Basic XXXYYYY`. 
+Notice that the authorization value is your OpenWhisk authorization string after being base64 encoded. 
+The response is of content type `application/json`.
+
+## Actions
+
+To create or update an action send a HTTP request with method `PUT` on the the actions collection. For example, to create a `nodejs:6` action with the name `hello` using a single file content use the following:
+```bash
+curl -u $AUTH -d '{"namespace":"_","name":"hello","exec":{"kind":"nodejs:6","code":"function main(params) { return {payload:\"Hello \"+params.name}}"}}' -X PUT -H "Content-Type: application/json" https://openwhisk.ng.bluemix.net/api/v1/namespaces/_/actions/hello?overwrite=true 
```

Comment:
Since we use POST to mean activate, we repurposed PUT to have create only behavior by default with the overwrite flag to allow the create or update behavior. More on this here https://github.com/openwhisk/openwhisk/issues/1171